packages/ERTP/test/unitTests/test-amountProperties.js

```diff
@@ -0,0 +1,79 @@
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import { makeCopyBag } from '@agoric/store';
+import fc from 'fast-check';
+
+import { AmountMath as m, AssetKind } from '../../src/index.js';
+import { mockBrand } from './mathHelpers/mockBrand.js';
+
+// Perhaps makeCopyBag should coalesce duplicate labels, but for now, it does not.
+const distinctLabels = pairs =>
+  new Set(pairs.map(([label, _qty]) => label)).size === pairs.length;
+const positiveCounts = pairs =>
+  pairs.filter(([_l, qty]) => qty > 0n).length === pairs.length;
+const arbBagContents = fc
+  .nat(7)
+  .chain(size =>
+    fc.array(
+      fc.tuple(fc.string(), fc.bigUint({ max: 1_000_000_000_000_000n })),
+      { minLength: size, maxLength: size },
+    ),
+  )
+  .filter(pairs => distinctLabels(pairs) && positiveCounts(pairs));
+
+const arbAmount = arbBagContents.map(contents =>
+  m.make(mockBrand, harden(makeCopyBag(contents))),
+);
+
+test('isEqual is a (total) equivalence relation', t => {
+  fc.assert(
+    fc.property(
+      fc.record({ x: arbAmount, y: arbAmount, z: arbAmount }),
+      ({ x, y, z }) => {
+        t.true([true, false].includes(m.isEqual(x, y))); // Total
+        t.true(m.isEqual(x, x)); // Reflexive
+        // Note: we write P => Q as !P || Q since JS has no logical => operator
```

Comment:
Rather than having to remember to reinterpret `!P || Q` as `P => Q` in my head, In #4389 I find that using an `implies` helper function makes things more readable. Do you agree?

Reply:
it's about the same to me. I accepted the suggestion, in any case.